index.ts

```diff
@@ -94,6 +96,51 @@ export class JolocomSDK {
     }
   }
 
+  /**
+   * Parses a recieved interaction token in JWT format and process it through
+   * the interaction system, returning the corresponding Interaction
+   *
+   * @param jwt recieved jwt string
+   * @returns Promise<Interaction> the associated Interaction object
+   */
+  public async processJWT(jwt: string): Promise<Interaction> {
```

Comment:
A very minor suggestion but not necessary, the `tokenRecieved` method above could be simplified a lot by using `processJWT` internally (just a bit of duplicated code).

Reply:
actually, I've now realised, `processJWT` doesn't indicate if the JWT is semantically valid. It would be nice if we had perhaps an error being thrown in the interaction flows when a semantic rule is violated (either checking the result of `processInteractionToken` or from within the flows themselves).

Reply:
`interaction.processInteractionToken` already throws a bunch of errors, and I wrappen them in an AppError now.

Reply:
And in the process I also fixed up `processInteractionToken` a bit, moved out the CredentialRecieve specific code to the CredentialOfferFlow
It wasn't actually doing anything with the computed validity array :D